firefox-ios/Client/Frontend/Share/URLActivityItemProvider.swift

```diff
@@ -0,0 +1,39 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import UniformTypeIdentifiers
+
+class URLActivityItemProvider: UIActivityItemProvider, @unchecked Sendable {
+    private let url: URL
+
+    static let excludedActivities = [
+        UIActivity.ActivityType.addToReadingList,
+    ]
+
+    init(url: URL) {
+        self.url = url
+
+        super.init(placeholderItem: url)
+    }
+
+    override var placeholderItem: Any? {
+        return url
+    }
+
+    override var item: Any {
+        if let activityType = activityType, URLActivityItemProvider.excludedActivities.contains(activityType) {
+            return NSNull()
+        }
```

Comment:
nit; Could be a `guard` instead of an `if`?

Reply:
I considered that, but we only want to return NSNull for non-nil activities that are excluded, not other non-nil activities (or even nil activities), so I thought it best to leave it as an if since it's the one exception case rather than a "bad path". 🤔